scripts/version updates/update_meilisearch_version.sh

```diff
@@ -0,0 +1,295 @@
+PENDING_LABEL="${BYELLOW}pending: ${NC}"
+INFO_LABEL="${BBLUE}info: ${NC}"
+
+## Utils Functions 
+
+# Rollback to previous MeiliSearch version in case something went wrong
+previous_version_rollback() {
+  echo "${ERROR_LABEL}MeiliSearch update to $meilisearch_version failed." >&2
+  echo "${INFO_LABEL}Rollbacking to previous version ${BPINK}$current_meilisearch_version${NC}." >&2
+  echo "${INFO_LABEL}Recovering..." >&2
+  mv /tmp/meilisearch /usr/bin/meilisearch
+  echo "${SUCCESS_LABEL}Recover previous data.ms." >&2
+  mv /tmp/data.ms /var/lib/meilisearch/data.ms
+  echo "${INFO_LABEL}Restarting MeiliSearch." >&2
+  systemctl restart meilisearch
+  systemctl_status exit
+  echo "${SUCCESS_LABEL}Previous MeiliSearch version ${BPINK}$current_meilisearch_version${NC} restarted correctly with its data recovered." >&2
+  delete_temporary_files
+  echo "${ERROR_LABEL}Update MeiliSearch from ${BPINK}$current_meilisearch_version${NC} to ${BPINK}$meilisearch_version${NC} failed. Rollback to previous version successfull." >&2
+  exit
+}
+
+# Check if MeiliSearch systemctl process is Active
+systemctl_status() {
+  systemctl status meilisearch | grep -E 'Active: active \(running\)' -q 
+  grep_status_code=$?
+  callback_1=$1
+  callback_2=$2
+  if [ $grep_status_code -ne 0 ]
+  then
+    echo "${ERROR_LABEL}MeiliSearch Service is not Running. Please start MeiliSearch." >&2
+    if [ ! -z "$callback_1" ]
+    then 
+      $callback_1
+    fi
+
+    if [ ! -z "$callback_2" ]
+    then 2
+      $callback_2
+    fi
+    
+  fi
+}
+
+# Delete temporary files
+delete_temporary_files() {
+  echo "${INFO_LABEL}Cleaning temporary files..."
+  if [ -f "meilisearch" ] ; then
+      rm meilisearch
+      echo "${SUCCESS_LABEL}Delete temporary meilisearch binary."
+  fi
+  
+  if [ -f "logs" ] ; then
+      rm logs
+      echo "${SUCCESS_LABEL}Delete temporary logs file."
+  fi
+
+  local dump_file="/dumps/$dump_id.dump"
+  if [ -f $dump_file ] ; then
+      rm "$dump_file"
+      echo "${SUCCESS_LABEL}Delete temporary dump file."
+  fi
+
+}
+
+# Check if MeiliSearch arguments are provided
+check_args () {
+  if [ $1 -eq 0 ]
+  then
+    echo "${ERROR_LABEL}$1"
+    exit
+  fi
+}
+
+# Check if latest command exit status is not an error
+check_last_exit_status () {
+  status=$1
+  message=$2
+  callback_1=$3
+  callback_2=$4
+
+  if [ $status -ne 0 ]
+  then
+    echo "${ERROR_LABEL}$messag."
+    if [ ! -z "$callback_1" ]
+    then 
+      ($callback_1)
+    fi
+    if [ ! -z "$callback_2" ]
+    then 
+      ($callback_2)
+    fi
+    exit
+  fi
+}
+
+
+
+## Main Script
+
+#
+# Current Running MeiliSearch 
+#
+
+echo "${SUCCESS_LABEL}Starting version update of MeiliSearch."
+
+# Check if MeiliSearch Service is running
+systemctl_status exit
+
+# Check if version argument was provided on script launch
+check_args $# "MeiliSearch Version not provided as arg.\nUsage: sh update_release.sh [vX.X.X]"
+
+# MeiliSearch update version.
+meilisearch_version=$1
```

Comment:
Maybe this line should be just before the functions?
The first function is `previous_version_rollback()` and it uses this variable, so if you try to read the script it could be better for the understanding.

Reply:
I think it is better to define the variable once we are sure an arg has been provided. Cause if I declare it higher `$1` might not exist yet. Thats why we check_args juste before it